### ISSUE TITLE
Existing Secrets Manager not found

### ISSUE THREAD

**Issue by in-1911** [terraform-ibm-modules/terraform-ibm-devsecops-alm#448]
When the prereq module is called (e.g. when `create_icr_namespace = true` ), the RG ID passed to it is supposed to be the toolchain's RG. Then the prereq module tries to look up secrets manager using the same RG, which fails if the secrets manager is in a different RG, which should be taken from the `sm_resource_group` variable at the root module.

### Affected modules

main.tf
prereq/main.tf

### Terraform CLI and Terraform provider versions

* Terraform version:
* Provider version:

### Debug output

```
2024/07/24 21:09:39 Terraform plan |   with module.prereqs[0].data.ibm_resource_instance.sm_instance[0],
 2024/07/24 21:09:39 Terraform plan |   on ../prereqs/main.tf line 23, in data "ibm_resource_instance" "sm_instance":
 2024/07/24 21:09:39 Terraform plan |   23: data "ibm_resource_instance" "sm_instance" {
 2024/07/24 21:09:39 Terraform plan | 
 2024/07/24 21:09:39 Terraform plan | ---
 2024/07/24 21:09:39 Terraform plan | id: terraform-23d5c674
 2024/07/24 21:09:39 Terraform plan | summary: |-
 2024/07/24 21:09:39 Terraform plan |   [ERROR] No resource instance found with name [rag-common-security-services-secrets-manager]
 2024/07/24 21:09:39 Terraform plan |   If not specified please specify more filters like resource_group_id if instance doesn't exists in default group, location or service
 2024/07/24 21:09:39 Terraform plan | severity: error
 2024/07/24 21:09:39 Terraform plan | resource: (Data) ibm_resource_instance
 2024/07/24 21:09:39 Terraform plan | operation: read
 2024/07/24 21:09:39 Terraform plan | component:
 2024/07/24 21:09:39 Terraform plan |   name: github.com/IBM-Cloud/terraform-provider-ibm
 2024/07/24 21:09:39 Terraform plan |   version: 1.67.1
 2024/07/24 21:09:39 Terraform plan | ---
```

### Steps to reproduce (including links and screen captures)

1. Run `terraform apply`

### Anything else

---

By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/terraform-ibm-modules/documentation/blob/main/CODE_OF_CONDUCT.md)

**Comment by huayuenh** [comment]
see https://github.com/terraform-ibm-modules/terraform-ibm-devsecops-alm/pull/450

**Comment by in-1911** [comment]
Any chance 1.9.5 will be pushed to the catalog (even as a pre-release) soon?

**Comment by padraic-edwards** [comment]
Fixed in 1.9.5
